### DISCUSSION TITLE
Vercel Serverless Function (Next.js API Handler) 504: Gateway time-out

### DISCUSSION THREAD

**Discussion by BluSimmon** [prisma/prisma#21374] [Q&A] [answered]
### Bug description

I use Prisma with MongoDB.
Everything on my local test is fine.
But when I host the website to Vercel.
The API function will get timeout by reaching 10s execution duration limit.
And after debug, I found the problem with the code below:
```ts
const user = await prisma.user.findFirst({
    where: {
      OR: [
        {
          username: {
            equals: username,
            mode: 'insensitive'
          }
        },
        {
          email: {
            equals: username,
            mode: 'insensitive'
          }
        },
      ]
    },
    select: {
      username: true,
      email: true,
      password: true,
      role: true,
    }
  })
```

### How to reproduce

You can have a test with https://test-prisma-login.vercel.app/
(You can input fields freely)

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  // MongoDB: ZunicMC > ZunicWeb
  provider = "mongodb"
  url      = env("DATABASE_URL")
}

model User {
  id       String   @id @default(auto()) @map("_id") @db.ObjectId
  username  String  @unique
  email     String  @unique
  password  String  
  role      Role    @default(USER)
  profile   Profile?
}

model Profile {
  id     String   @id @default(auto()) @map("_id") @db.ObjectId
  user    User    @relation(fields: [userId], references: [id])
  userId  String  @db.ObjectId @unique
}

enum Role {
  USER
  ADMIN
  OWNER
}
```

### Environment & setup

- OS (development): Windows
- Database: MongoDB
- Node.js version: v18.16.1

### Prisma Version

```
prisma                  : 5.3.1
@prisma/client          : 5.3.1
Current platform        : windows
Query Engine (Node-API) : libquery-engine 61e140623197a131c2a6189271ffee05a7aa9a59 (at node_modules\@prisma\engines\query_engine-windows.dll.node)
Schema Engine           : schema-engine-cli 61e140623197a131c2a6189271ffee05a7aa9a59 (at node_modules\@prisma\engines\schema-engine-windows.exe)
Schema Wasm             : @prisma/prisma-schema-wasm 5.3.1-2.61e140623197a131c2a6189271ffee05a7aa9a59
Default Engines Hash    : 61e140623197a131c2a6189271ffee05a7aa9a59
Studio                  : 0.494.0
```

**Comment by janpio** [comment]
What is the problem you found in that code?
Does other code work and just this query is broken?
Where are you hosting your MongoDB instance?

**Comment by BluSimmon** [comment]
The code execute with a long time, it reach the execution duration limit of Vercel plan.
Everything is work, just the execution time is out of limit.
I host the MongoDB with MongoDB Cloud.

**Reply by janpio** [reply]
That sounds like your MongoDB database might not actually be accessible from the Vercel function.
Are your sure that your database should be accessible from that function?

**Reply by BluSimmon** [reply]
The project's environment variables and schema database connection url are setted correctly.
But there isn't any error that i can see from the  build log.
This is all I can see from vercel:
![image](https://github.com/prisma/prisma/assets/57604186/03f693f7-107e-4aee-8417-19455ad09a66)

**Reply by janpio** [reply, marked as answer]
Do you have to whitelist the requesting IP on the MongoDB side?
No logs on Vercel?

**Reply by BluSimmon** [reply]
Oh, it is sorry that for my mistake.
The solution is to allow access from everywhere from MongoDB.
Thank you for your helps!

**Reply by doghousetv** [reply]
This discussion post just saved me, thank you so much

**Reply by aaron-mota** [reply]
Same!  Created a new MongoDB project/database for my latest NextJS/T3 stack/Vercel project, and I think MongoDB changed the setup flow because I didn't see the Network Access screen where I typically remember to allow access from everywhere (0.0.0.0/0).

Glad I came across this 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!

**Comment by maharjanm96** [comment]
use mongo db cloud and allow network access from all IP , it worked for me
